src/rust/engine/src/rule_graph.rs

```diff
@@ -624,21 +600,18 @@ impl<'t> GraphMaker<'t> {
   ///
   fn choose_dependencies<'a>(
     available_params: &ParamTypes,
-    deps: &'a [(SelectKey, Vec<Entry>)],
-  ) -> Result<Option<Vec<(&'a SelectKey, &'a Entry)>>, Diagnostic> {
+    deps: &'a [(DependencyKey, Vec<Entry>)],
+  ) -> Result<Option<Vec<(&'a DependencyKey, &'a Entry)>>, Diagnostic> {
     let mut combination = Vec::new();
     for (key, input_entries) in deps {
-      let provided_param = match &key {
-        &SelectKey::JustSelect(_) => None,
-        &SelectKey::JustGet(ref g) => Some(&g.subject),
-      };
+      let provided_param = key.provided_param();
       let satisfiable_entries = input_entries
         .iter()
         .filter(|input_entry| {
           input_entry
             .params()
             .iter()
-            .all(|p| available_params.contains(p) || Some(p) == provided_param)
+            .all(|p| available_params.contains(p) || Some(*p) == provided_param)
```

Comment:
Rust will dereference automatically, right? This is being done as a matter of explicitness?

Reply:
In some case it will, yea. I think that in the case the dereference is necessary in order to construct the right type of option for the comparison (ie, the auto-deref doesn't kick in, because it's totally valid to have an `Option<&TypeId>`... it's not what we want here.)